### PR TITLE
[Merged by Bors] - feat: port Order.Iterate

### PR DESCRIPTION
**Pull request by dwarn**
mathlib SHA: fd47bdf09e90f553519c712378e651975fe8c829

### CODE REVIEW

File: Mathlib/Order/Iterate.lean [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+In this file we prove some inequalities comparing `f^[n] x` and `g^[n] x` where `f` and `g` are
+two self-maps that commute with each other.
+
+Current selection of inequalities is motivated by formalization of the rotation number of
+a circle homeomorphism.
+-/
+
+
+variable {α β : Type _}
```
Comment (mcdoll) [inline Mathlib/Order/Iterate.lean:21]:
```suggestion
```
We can use autoimplicits here: https://github.com/leanprover-community/mathlib4/wiki#auto-implicits

File: Mathlib/Order/Iterate.lean [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+variable [LinearOrder α] {f g : α → α}
+
+theorem iterate_pos_lt_iff_map_lt (h : Commute f g) (hf : Monotone f) (hg : StrictMono g) {x n}
+    (hn : 0 < n) : (f^[n]) x < (g^[n]) x ↔ f x < g x := by
+  rcases lt_trichotomy (f x) (g x) with (H | H | H)
+  · simp only [*, iterate_pos_lt_of_map_lt]
+  · simp only [*, h.iterate_eq_of_map_eq, lt_irrefl]
+  · simp only [lt_asymm H, lt_asymm (h.symm.iterate_pos_lt_of_map_lt' hg hf H hn)]
+
+#align function.commute.iterate_pos_lt_iff_map_lt Function.Commute.iterate_pos_lt_iff_map_lt
```
Comment (mcdoll) [inline Mathlib/Order/Iterate.lean:199]:
```suggestion
#align function.commute.iterate_pos_lt_iff_map_lt Function.Commute.iterate_pos_lt_iff_map_lt
```

**Review by mcdoll — APPROVED** [review]
Looks good to me

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**